src/parser/view_transform.c

```diff
@@ -2758,75 +2757,36 @@ pt_pushable_query_in_pos (PARSER_CONTEXT * parser, PT_NODE * query, int pos)
 	PT_NODE *list;
 	int i;
 
+	cinfo.query_found = false;
+	cinfo.method_found = false;
+	cinfo.xxxnum_found = false;
+	cinfo.analytic_found = false;
 	/* Traverse select list */
 	for (list = query->info.query.q.select.list, i = 0; list; list = list->next, i++)
 	  {
-	    /* init */
-	    cinfo.check_query = (i == pos) ? true : false;
-	    cinfo.check_method = (i == pos) ? true : false;
-	    cinfo.check_xxxnum = true;	/* always check */
-	    cinfo.check_analytic = (i == pos) ? true : false;
-
-	    cinfo.query_found = false;
-	    cinfo.method_found = false;
-	    cinfo.xxxnum_found = false;
-	    cinfo.analytic_found = false;
-
+	    if (i != pos)
+	      {
+		continue;
+	      }
 	    switch (list->node_type)
 	      {
 	      case PT_SELECT:
 	      case PT_UNION:
 	      case PT_DIFFERENCE:
 	      case PT_INTERSECTION:
-		if (i == pos)
-		  {
-		    cinfo.query_found = true;	/* not pushable */
-		  }
+		cinfo.query_found = true;	/* not pushable */
 		break;
 
 	      case PT_METHOD_CALL:
-		if (i == pos)
-		  {
-		    cinfo.method_found = true;	/* not pushable */
-		  }
-		break;
-
-	      case PT_EXPR:
-		/* always check for rownum, inst_num(), orderby_num() */
-		if (list->info.expr.op == PT_ROWNUM || list->info.expr.op == PT_INST_NUM
-		    || list->info.expr.op == PT_ORDERBY_NUM)
-		  {
-		    cinfo.xxxnum_found = true;	/* not pushable */
-		  }
-		else
-		  {		/* do traverse */
-		    parser_walk_leaves (parser, list, pt_check_pushable, &cinfo, NULL, NULL);
-		  }
-		break;
-
-	      case PT_FUNCTION:
-		/* always check for groupby_num() */
-		if (list->info.function.function_type == PT_GROUPBY_NUM)
-		  {
-		    cinfo.xxxnum_found = true;	/* not pushable */
-		  }
-		else if (list->info.function.analytic.is_analytic == true)
-		  {
-		    cinfo.analytic_found = true;	/* not pushable */
-		  }
-		else
-		  {		/* do traverse */
-		    parser_walk_leaves (parser, list, pt_check_pushable, &cinfo, NULL, NULL);
-		  }
+		cinfo.method_found = true;	/* not pushable */
 		break;
 
 	      default:		/* do traverse */
 		parser_walk_leaves (parser, list, pt_check_pushable, &cinfo, NULL, NULL);
 		break;
 	      }			/* switch (list->node_type) */
 
-	    /* check for subquery, method, rownum, inst_num(), orderby_num(), groupby_num(): does not pushable if we
-	     * find these in corresponding item in select_list of query */
+	    /* check for subquery, method does not pushable in select_list of subquery */
 	    if (cinfo.query_found || cinfo.method_found || cinfo.xxxnum_found || cinfo.analytic_found)
 	      {
 		break;		/* not pushable */
```

Comment:
Since it is a structure that only works in the case of  (i == pos) , this loop can reach this point only once.

```
if (cinfo.query_found || cinfo.method_found || cinfo.xxxnum_found || cinfo.analytic_found)
    {
		break;		/* not pushable */
    }
else 
   {
      list = NULL;
      break;
```

Reply:
Okay. I'll fix it.